kitsune/users/views.py

```diff
@@ -759,3 +773,107 @@ def get_settings(attr, *args):
         if val is not None:
             return val
         return super(FXALogoutView, FXALogoutView).get_settings(attr, *args)
+
+
+class WebhookView(View):
+    """The flow here is based in the mozilla-django-oidc lib.
+
+    If/When the said lib supports SET tokens this will be replaced by the lib.
+    """
+    @staticmethod
+    def get_settings(attr, *args):
+        """Override the logout url for Firefox Accounts."""
+
+        val = get_oidc_fxa_setting(attr)
+        if val is not None:
+            return val
+        return import_from_settings(attr, *args)
+
+    def verify_token(self, token, **kwargs):
+        """Validate the token signature."""
+
+        token = force_bytes(token)
+        key = self.OIDC_RP_CLIENT_SECRET
+
+        jws = JWS.from_compact(token)
```

Comment:
Replace this with an OIDCBackend instance and call the methods instead of copy/paste-ing code.